sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -932,6 +933,33 @@ object PushPredicateThroughGenerate extends Rule[LogicalPlan] with PredicateHelp
 }
 
 /**
+ * Push [[Filter]] operators through [[Window]] operators. Parts of the predicate that can be pushed
+ * beneath must satisfy three conditions:
+ * 1. involving one and only one column that is part of window partitioning key.
+ * 2. Window partitioning key should be just a sequence of [[AttributeReference]].
```

Comment:
I am just reading the way how Hive did the work. Actually, this is also my question.

https://issues.apache.org/jira/secure/attachment/12788757/HIVE-12808.05.patch

In this patch, you can see there are test cases to avoid this case. 
`Test predicate is not getting pushed down when window has compound partition key`